drivers/spi/spi_mcux_flexcomm.c

```diff
@@ -88,6 +89,11 @@ static void spi_mcux_transfer_next_packet(struct device *dev)
 		transfer.configFlags = kSPI_FrameAssert;
 	}
 
+	if (ctx->rx_len && !transfer.rxData) {
+		// workaround: fsl_spi.c will set rx_len = 0 if rxData == NULL
+		transfer.rxData = &dummy;
```

Comment:
Will not this be a buffer issue then?  it will work if rx_len is 1, but not if it is more.

Reply:
@tbursztyka there are cases where we rx the CRC which is 4 bytes long.
It does not crash as it's the last variable in the stack and the hal driver allocates not new heap memory.

Reply:
That is still bad. What are the guaranty that rx_len is never going to be >= 4?

@MaureenHelm Could the HAL be fixed? I mean, if the user does not want to read anything then both rxData and rx_len are 0, fine. But some time it may want to "read & forget" some rx data, thus why the rx buffer  is NULL but not the rx_len.

Reply:
@tbursztyka discarded this pull, made a fix in hal_nxp